scout_apm.gemspec

```
@@ -28,7 +28,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency "rake-compiler"
   s.add_development_dependency "addressable"
   s.add_development_dependency "activesupport"
-  s.add_runtime_dependency "parser"
+  s.add_runtime_dependency "parser", Gem::Version.new(RUBY_VERSION).approximate_recommendation
```

Per https://bundler.io/guides/gemfile.html

> Most of the version specifiers, like >= 1.0, are self-explanatory. The specifier ~> has a special meaning, best shown by example. ~> 2.0.3 is identical to >= 2.0.3 and < 2.1. ~> 2.1 is identical to >= 2.1 and < 3.0. ~> 2.2.beta will match prerelease versions like 2.2.beta.12. ~> 0 is identical to >= 0.0 and < 1.0.

```suggestion
  s.add_runtime_dependency "parser", "#{Gem::Version.new(RUBY_VERSION).approximate_recommendation}.0"
```

Since Parser has a version that corresponds to every minor version, we want `~> 3.2.0` which means `>= 3.2.0 and < 3.3`. Right now you've got `~> 3.2` which means `>= 3.2 and < 4.0`. But I don't think we want to use Parser 3.3 on Ruby 3.2.

---

Good catch, thanks!

---

Hmm getting these errors for the tests

```
warning: parser/current is loading parser/ruby23, which recognizes
warning: 2.3.3-compliant syntax, but you are running 2.3.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
rake aborted!
LoadError: Parser::TreeRewriter was not defined
```

```
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.10.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
rake aborted!
LoadError: Parser::TreeRewriter was not defined
```

[The last Parser 2.3 is Parser 2.3.3.1](https://github.com/whitequark/parser/blob/master/CHANGELOG.md). Looks like we're expected to use Parser 2.4 for Ruby 2.3.8.

> But I don't think we want to use Parser 3.3 on Ruby 3.2.

Agree with you that this is not ideal, but it's probably safe enough. The main thing is to get folks to upgrade the parser whenever they upgrade their Ruby version.

Gonna revert for now.

---

Thanks for the continued effort on this guys. We appreciate it!

I think this is good.

This also may solve some issues related to:
https://github.com/scoutapp/scout_apm_ruby/issues/445

But still allows people below Ruby 2.5 to get to a parser version above 2.5 which has the TreeRewriter. Again also not ideal, but probably safe